Source/WebCore/style/Styleable.cpp

```diff
@@ -194,6 +195,11 @@ bool Styleable::isRunningAcceleratedTransformAnimation() const
 
 bool Styleable::runningAnimationsAreAllAccelerated() const
 {
+    if (auto* renderer = dynamicDowncast<RenderLayerModelObject>(this->renderer())) {
+        if (renderer->hasAcceleratedEffects())
```

Comment:
We only use threaded animations when all animations on an element are accelerated? Where is this decision made?

Reply:
I'm also a little surprised that any change is needed here. Why aren't threaded animations represented in `keyframeEffectStack()` which would make the below code take care of this?

Reply:
> We only use threaded animations when all animations on an element are accelerated? Where is this decision made?

No, we don't. We use threaded animations as long as properties set on that element doesn't prevent us from doing so. Much of that code hasn't landed yet as I still need to refactor it a bit but some examples would be:

- animating a width-dependent `translate` value while also animating `width`
- animating `filter` with a discrete interval yielding mis-matching filter lists
- animating certain Motion Path properties which require computations we cannot perform in the UI Process

I don't recall why this method is named the way it is since the single call site for it `RenderBoxModelObject::hasRunningAcceleratedAnimations()` which certainly does not sound like it's exclusive. The only situation where we should be interested in whether **all** animations are running accelerated is when we want to avoid scheduling animations in the Web Process.

> I'm also a little surprised that any change is needed here. Why aren't threaded animations represented in `keyframeEffectStack()` which would make the below code take care of this?

The way the threaded animation resolution code works is that we create a separate `AcceleratedEffectStack` containing accelerated effects that will get uploaded to the UIProcess. The managed objects are different, `AcceleratedEffect` v. `KeyframeEffect`. They share some traits, for instance both support resolving timing and blending keyframes, but they are different since one is designed to animate `CALayer` properties and the other CSS properties. One is designed to have only the minimal number of properties to run in the UI Process, the other support all the complexity of the Web Animations API.

The code below is really the "old way" where calling `Keyframe::isRunningAccelerated()` would indicate whether a `CAAnimation` is running for that effect. The new code queries the accelerated effect stack to see if it exists and has effects, meaning there are animations running in the UI Process for it.